units/pod_creation_test.go

```diff
@@ -0,0 +1,197 @@
+			require.NoError(t, j.pod.Insert())
+			require.NoError(t, j.pod.UpdateStatus(pod.StatusRunning))
+			assert.Equal(t, pod.StatusRunning, j.pod.Status)
+
+			j.Run(ctx)
+			require.Error(t, j.Error())
+
+			dbPod, err := pod.FindOneByID(j.PodID)
+			require.NoError(t, err)
+			require.NotZero(t, dbPod)
+			assert.Equal(t, pod.StatusRunning, dbPod.Status)
+		},
+		"FailsWithTerminatedStatus": func(ctx context.Context, t *testing.T, j *createPodJob) {
+			require.NoError(t, j.pod.Insert())
+			require.NoError(t, j.pod.UpdateStatus(pod.StatusTerminated))
+
+			j.Run(ctx)
+			require.Error(t, j.Error())
+
+			dbPod, err := pod.FindOneByID(j.PodID)
+			require.NoError(t, err)
+			require.NotZero(t, dbPod)
+			assert.Equal(t, pod.StatusTerminated, dbPod.Status)
+		},
+		"FailsWhenExportingOpts": func(ctx context.Context, t *testing.T, j *createPodJob) {
+			j.pod.TaskContainerCreationOpts.Image = ""
+			require.NoError(t, j.pod.Insert())
+
+			j.Run(ctx)
+			require.Error(t, j.Error())
+			dbPod, err := pod.FindOneByID(j.PodID)
+			require.NoError(t, err)
+			require.NotZero(t, dbPod)
+			assert.Equal(t, pod.StatusInitializing, dbPod.Status)
+		},
+		"FailsWhenCreatingPod": func(ctx context.Context, t *testing.T, j *createPodJob) {
+			j.pod.TaskContainerCreationOpts.CPU = 0
+			require.NoError(t, j.pod.Insert())
+
+			j.Run(ctx)
+			require.Error(t, j.Error())
+
+			dbPod, err := pod.FindOneByID(j.PodID)
+			require.NoError(t, err)
+			require.NotZero(t, dbPod)
+			assert.Equal(t, pod.StatusInitializing, dbPod.Status)
+		},
+	} {
+		t.Run(tName, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			require.NoError(t, db.ClearCollections(pod.Collection, event.AllLogCollection))
+			defer func() {
+				assert.NoError(t, db.ClearCollections(pod.Collection, event.AllLogCollection))
+			}()
+
+			cluster := "cluster"
+			cocoaMock.GlobalECSService.Clusters[cluster] = cocoaMock.ECSCluster{}
+
+			defer func() {
+				cocoaMock.GlobalECSService = cocoaMock.ECSService{
+					Clusters: map[string]cocoaMock.ECSCluster{},
+					TaskDefs: map[string][]cocoaMock.ECSTaskDefinition{},
+				}
+				cocoaMock.GlobalSecretCache = map[string]cocoaMock.StoredSecret{}
+			}()
+
+			p := pod.Pod{
+				ID:     "id",
+				Status: pod.StatusInitializing,
+				TaskContainerCreationOpts: pod.TaskContainerCreationOptions{
+					Image:    "image",
+					MemoryMB: 128,
+					CPU:      128,
+					OS:       pod.OSLinux,
+					Arch:     pod.ArchAMD64,
+					EnvSecrets: map[string]string{
+						"s0": utility.RandomString(),
+						"s1": utility.RandomString(),
+					},
+				},
+			}
+
+			env := evgMock.Environment{}
+			require.NoError(t, env.Configure(ctx))
+			var envClusters []evergreen.ECSClusterConfig
+			for name := range cocoaMock.GlobalECSService.Clusters {
+				envClusters = append(envClusters, evergreen.ECSClusterConfig{
+					Name: name,
```

Comment:
You have to set the platform to match the pod's OS (linux) if you want the test to successfully match the pod OS with the ECS cluster.